src/main/java/org/tron/core/services/RpcApiService.java

```diff
@@ -768,12 +984,43 @@ public void getNowBlock(EmptyMessage request, StreamObserver<Block> responseObse
       responseObserver.onCompleted();
     }
 
+    @Override
+    public void getNowBlock2(EmptyMessage request,
+        StreamObserver<BlockExtention> responseObserver) {
+      Block block = wallet.getNowBlock();
+      responseObserver.onNext(block2Extention(block));
+      responseObserver.onCompleted();
+    }
+
     @Override
     public void getBlockByNum(NumberMessage request, StreamObserver<Block> responseObserver) {
       responseObserver.onNext(wallet.getBlockByNum(request.getNum()));
       responseObserver.onCompleted();
     }
 
+    @Override
+    public void getBlockByNum2(NumberMessage request,
+        StreamObserver<BlockExtention> responseObserver) {
+      Block block = wallet.getBlockByNum(request.getNum());
+      responseObserver.onNext(block2Extention(block));
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getTransactionCountByBlockNum(NumberMessage request,
+        StreamObserver<NumberMessage> responseObserver) {
+      Block block = null;
+      try {
+        block = dbManager.getBlockByNum(request.getNum()).getInstance();
+      } catch (StoreException e) {
+        logger.error(e.getMessage());
+      }
+      NumberMessage.Builder builder = NumberMessage.newBuilder();
+      builder.setNum(block.getTransactionsCount());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "block" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1180&issues=AWTWJrjdVtLgEGHvwB6H&open=AWTWJrjdVtLgEGHvwB6H)

[ISSUE_KEY]: AWTWJrjdVtLgEGHvwB6H